client/my-sites/stats/stats-purchase/stats-purchase-tier-upgrade-slider.tsx

```diff
@@ -0,0 +1,92 @@
+import { PricingSlider } from '@automattic/components';
+import React, { useState } from 'react';
+import { COMPONENT_CLASS_NAME } from './stats-purchase-wizard';
+
+function getPlanTiers() {
+	// Pulled from the API/Redux.
+	const planTiers = [
+		{
+			id: '10k',
+			price: '$9',
+			views: '10k',
+			description: '$9/month for 10k views',
+		},
+		{
+			id: '100k',
+			price: '$19',
+			views: '100k',
+			description: '$19/month for 100k views',
+		},
+		{
+			id: '250k',
+			price: '$29',
+			views: '250k',
+			description: '$29/month for 250k views',
+		},
+		{
+			id: '500k',
+			price: '$49',
+			views: '500k',
+			description: '$49/month for 500k views',
+		},
+		{
+			id: '1M',
+			price: '$69',
+			views: '1M',
+			description: '$69/month for 1M views',
+		},
+		{
+			id: '1M++',
+			price: '$69++',
+			views: '1M++',
+			description: '$25/month per million views if views exceed 1M',
+		},
+	];
+
+	return planTiers;
+}
+
+function TierUpgradeSlider() {
+	// Get the plan details.
+	const plans = getPlanTiers();
+
+	// Adjust available plans based on current tier.
+	const availablePlans = plans;
+
+	// Slider state.
+	const [ currentPlanIndex, setCurrentPlanIndex ] = useState( 0 );
+	const sliderMin = 0;
+	const sliderMax = plans.length - 1;
+
+	const handleSliderChange = ( value: number ) => {
+		// console.log( 'slider did change. value: ', value );
+		// console.log( 'plan info: ', plans[value] );
+		setCurrentPlanIndex( value );
```

Comment:
It's ok for now, but if the final source will have IDs, it will be better to find the object based on that rather than the array index.